File: array/170.py
from typing import List


def missingNumber(nums: List[int]) -> int:
    sum = 0
    n = len(nums)

    for elem in nums:
        sum += elem

    return n * (n + 1) // 2 - sum


print(missingNumber([9, 6, 4, 2, 3, 5, 7, 0, 1]))

File: array/170.cpp
#include <bits/stdc++.h>

using namespace std;

int missingNumber(vector<int>& nums) {
    int sum = 0;
    int n = nums.size();
    for(const auto & elem: nums)
        sum += elem;

    return n*(n+1)/2 - sum;
}

int main() {
    vector<int> nums = {9,6,4,2,3,5,7,0,1};
    cout<<missingNumber(nums);


    return 0;
}